vehicle/src/Vehicle/Compile/Type/Monad/Class.hs

```diff
@@ -365,7 +360,8 @@ prettyMetaInternal m t = pretty m <+> ":" <+> prettyVerbose t
 clearMetaCtx :: forall builtin m. (MonadTypeChecker builtin m) => Proxy builtin -> m ()
 clearMetaCtx _ = do
   logDebug MaxDetail "Clearing meta-variable context"
-  modifyTypeCheckerState @builtin $ const emptyTypeCheckerState
+  instanceDatabase <- instanceCandidates <$> getTypeCheckerState @builtin
+  modifyTypeCheckerState @builtin $ const emptyTypeCheckerState {instanceCandidates = instanceDatabase}
```

Comment:
Could probably write this more efficiently as: 
```haskell
modifyTypeCheckerState @builtin $ \state -> 
   emptyTypeCheckerState {instanceCandidates = instanceCandidates state}
```
(haven't type checked this snippet). That would avoid us getting and then modifying the state!